services/tone_analyzer/v3.js

```diff
@@ -0,0 +1,186 @@
+  // Otherwise, once set username would never get reset, resulting in a frustrated
+  // user who, when he errenously enters bad credentials, can't figure out why
+  // the edited ones are not being taken.
+
+  // Not ever used, and codeacy complains about it.
+
+  var username, password, sUsername, sPassword;
+
+  var service = cfenv.getAppEnv().getServiceCreds(/tone analyzer/i)
+
+  if (service) {
+    sUsername = service.username;
+    sPassword = service.password;
+  }
+
+  // Node RED Admin - fetch and set vcap services
+  RED.httpAdmin.get('/watson-tone-analyzer/vcap', function (req, res) {
+    res.json(service ? {bound_service: true} : null);
+  });
+
+  // Function that checks the payload and determines
+  // whether it is JSON or a Buffer
+  var checkPayload = function(payload) {
+    var message = null;
+    var isBuffer = false;
+
+    var hasJSONmethod = (typeof payload.toJSON === 'function') ;
+
+    if (hasJSONmethod === true) {
+      if (payload.toJSON().type === 'Buffer') {
+        isBuffer = true;
+      }      
+    }      
+    // Payload (text to be analysed) must be a string (content is either raw string or Buffer)
+    if (typeof payload !== 'string' &&  isBuffer !== true) {
+      message = 'The payload must be either a string or a Buffer';
+    }
+
+    return message;
+  };
+
+  // Check that the credentials have been provided
+  // Credentials are needed for each the service.
+  var checkCreds = function(credentials) {
+    var taSettings = null;
+
+    username = sUsername || credentials.username;
+    password = sPassword || credentials.password;      
+
+    if (username && password) {
+      taSettings = {};
+      taSettings.username = username;
+      taSettings.password = password;
+    }
+
+    return taSettings;
+  }
+
+
+  // Function that checks the configuration to make sure that credentials,
+  // payload and options have been provided in the correct format.
+  var checkConfiguration = function(msg, node, cb) {
+    var message = null;      
+    var taSettings = null;
+
+    taSettings = checkCreds(node.credentials);
+
+    if (!taSettings) {
+      message = 'Missing Tone Analyzer service credentials';
+    } else if (!msg.payload) {
+      message = 'Missing property: msg.payload';
+      if (!message) {
+        message = checkPayload(msg.payload);
+      }
+    }     
+
+    if (cb) {
+      cb(message, taSettings);
+    }
+  };
+
+  // function to parse through the options in preparation
+  // for the sevice call.
+  var parseOptions = function (msg, config) {
+    var tones = msg.tones || config.tones;
+    var sentences = msg.sentences || config.sentences;
+    var contentType = msg.contentType || config.contentType
+
+    var options = {
+      'text': msg.payload,
+      'sentences': sentences,   
+      'isHTML': contentType     
+    };
+
+    // if (tones !== 'all') {
+    //  options.tones = tones;
+    // }
+
+    switch (tones) {
+      case 'all':
+        options.tones = tones
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Expected indentation of 6 space characters but found 8.](https://www.codacy.com/app/rezgui-y/node-red-node-watson/file/2658175860/issues/source?bid=3225115&fileBranchId=3326638#l125)